### PR TITLE
_parse_base_groups said fixme refactor. so, why not?

### PR DESCRIPTION
**Pull request by skyl**
I was just learning and exploring. It said `FIXME: refactor`. So, I did.

### CODE REVIEW

File: lib/ansible/inventory/expand_hosts.py [hunk 1]
```diff
@@ -34,7 +34,8 @@
 
 from ansible import errors
 
-def detect_range(line = None):
```
Comment (skyl) [inline lib/ansible/inventory/expand_hosts.py:37]: `None` is no good as a default.

File: lib/ansible/inventory/expand_hosts.py [hunk 2]
```diff
@@ -64,6 +66,8 @@ def expand_hostname_range(line = None):
     '''
     all_hosts = []
     if line:
+        if not detect_range(line):
```
Comment (skyl) [inline lib/ansible/inventory/expand_hosts.py:69]: with this, the caller can just `expand_hostname_range(line)` without having to guard against it not being in the range formate.

File: lib/ansible/inventory/ini.py [hunk 3]
```diff
@@ -55,79 +66,23 @@ def _parse(self):
     # delta asdf=jkl favcolor=red
 
     def _parse_base_groups(self):
-        # FIXME: refactor
-
         ungrouped = Group(name='ungrouped')
-        all = Group(name='all')
```
Comment (skyl) [inline lib/ansible/inventory/ini.py:61]: `all` is a builtin.

File: lib/ansible/inventory/host.py [hunk 4]
```diff
@@ -37,6 +40,23 @@ def add_group(self, group):
 
         self.groups.append(group)
 
+    def set_variables_from_tokens(self, tokens):
```
Comment (skyl) [inline lib/ansible/inventory/host.py:43]: I think this is nice as a method on `Host`

File: lib/ansible/inventory/ini.py [hunk 5]
```diff
@@ -55,79 +66,23 @@ def _parse(self):
...
-                # Three cases to check:
-                # 0. A hostname that contains a range pesudo-code and a port
-                # 1. A hostname that contains just a port
-                if hostname.count(":") > 1:
-                    # probably an IPv6 addresss, so check for the format
-                    # XXX:XXX::XXX.port, otherwise we'll just assume no
-                    # port is set 
-                    if hostname.find(".") != -1:
-                        (hostname, port) = hostname.rsplit(".", 1)
-                elif (hostname.find("[") != -1 and
```
Comment (skyl) [inline lib/ansible/inventory/ini.py:96]: this logic was hard to read and understand. I think it's better as `hostname_has_port` function. It's possible that this function is more widely useful and should be moved somewhere else.

File: lib/ansible/inventory/ini.py [hunk 6]
```diff
@@ -55,79 +66,23 @@ def _parse(self):
...
-
-                hostnames = []
-                if detect_range(hostname):
-                    hostnames = expand_hostname_range(hostname)
-                else:
-                    hostnames = [hostname]
-
-                for hn in hostnames:
-                    host = None
-                    if hn in self.hosts:
```
Comment (skyl) [inline lib/ansible/inventory/ini.py:111]: `dict.setdefault` is my friend.

**Comment by mpdehaan** [comment]
In queue for review/testing, thanks!

**Comment by mpdehaan** [comment]
I realize this one has been stuck in queue for a while, we've had quite a backlog and have been prioritizing -- refactoring often is not always the highest priority.

Anyway, since things have changed a lot under the hood -- things no longer apply and we are quite interested in serge's speedups fixes on top, which will probably make his harder to apply.  As a result, I'm going to close this one.

I hope you understand.

Thank you!

**Pull request closed without merging**